### PR TITLE
PLAT-3397 - Advanced search for basic object fields

### PR DESCRIPTION
**Pull request by rkreich**

### CODE REVIEW

File: alpha/apps/kaltura/lib/model/objectfilters/AdvancedSearchFilterMatchAttributeCondition.class.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	/**
+	 * Adds conditions, matches and where clauses to the query
+	 * @param IKalturaDbQuery $query
+	 */
+	public function applyCondition(IKalturaDbQuery $query)
+	{
+		if (!$query instanceof IKalturaIndexQuery)
+			return;
+
+		$matchText = '"'.SphinxUtils::escapeString($this->value).'"';
```
Comment (tan-tan-kanarek) [inline alpha/apps/kaltura/lib/model/objectfilters/AdvancedSearchFilterMatchAttributeCondition.class.php]: Roman,
SphinxUtils is a plugin class.
Please consider moving the entire match functionality into search plugin, or, at least, generalize SphinxUtils::escapeString somehow.

File: alpha/lib/model/index/entryIndex.php [hunk 2]
```diff
@@ -378,6 +378,64 @@ public static function getSphinxConditionsToKeep()
...
 	}
 
+	protected static $apiCompareAttributesMap;
+
+	public static function getApiCompareAttributesMap()
+	{
+		if (!self::$apiCompareAttributesMap)
+		{
+			self::$apiCompareAttributesMap = array(
+				'status' => 'entry_status',
```
Comment (tan-tan-kanarek) [inline alpha/lib/model/index/entryIndex.php]: Roman,
Can't we add these definitions to the index XML?

Eran,
We need to add verification to the static code validator, could you remind us where it could be found?

File: api_v3/lib/types/filters/advanced/KalturaSearchComparableAttributeCondition.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+	/**
+	 * @var KalturaSearchConditionComparison
+	 */
+	public $comparison;
+
+	/**
+	 * Placeholder property, the real property is defined on parent classes
+	 */
+	protected $attribute;
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/filters/advanced/KalturaSearchComparableAttributeCondition.php]: Check how it looked in client libraries

File: plugins/tag_search/lib/model/index/TagIndex.php [hunk 4]
```diff
@@ -43,7 +43,7 @@ public static function getIndexFieldsMap()
 		{
 			self::$fieldsMap = array(
 				'int_id' => 'intId',
-				'tag' => 'TagWithEqual',
+				'tag' => 'tag',
```
Comment (tan-tan-kanarek) [inline plugins/tag_search/lib/model/index/TagIndex.php]: Roman, is that change on purpose?

@hilak, please review this change and approve.

File: alpha/lib/model/index/entryIndex.php (hunk 2 above)
Comment (rkreich) [inline alpha/lib/model/index/entryIndex.php]: What do you mean? This method is generated based on IndexSchema.xml

File: plugins/tag_search/lib/model/index/TagIndex.php (hunk 4 above)
Comment (rkreich) [inline plugins/tag_search/lib/model/index/TagIndex.php]: This is how it was generated, but I now see that TagWithEqual is a real getter, I'll see if i can modify IndexSchema.xml to generate the old output.
There was a similar issue in entry index, that 2 fields were added manually without updating IndexSchema.xml

Comment (rkreich) [inline plugins/tag_search/lib/model/index/TagIndex.php]: I don't see how I can align this using IndexSchema.xml
@hilak, do you know why it was implemented like that?

Comment (hilak) [inline plugins/tag_search/lib/model/index/TagIndex.php]: This was committed by Noam Arad. @noam-arad , could you please explain this implementation?
